Readme + Swiftlint + Tests

sergigracia opened the pull request:
- [x] Update Readme to add documentation
- [x] Update swift classes to remove Swiftlint warnings
- [x] Add tests for UnsplashKit
- [x] Add tests for UnsplashKit+Calls

sergigracia committed:
Update project to fix swiftlint warnings

sergigracia committed:
Update Readme

sergigracia committed:
Update testing project config

sergigracia committed:
Added specs for UnsplashKit class

sergigracia committed:
Completed all the UnsplashKit call specs

pepicrft commented on Example/Tests/Configuration/UnsplashKitSharedExample.swift:
```diff
@@ -0,0 +1,35 @@
+                        done()
+                    }
+                }
+                let stubbedImageData = UIImagePNGRepresentation(stubbedImage)!
```
You can even use `UIImage()` directly _(no need to add an image to the bundle)_

sergigracia replied:
Yep, I did that because I wanted to do a NSData comparison but it didn't work so I can remove it.

sergigracia replied:
If I use an empty UIImage then compare data length doesn't make sense so for now, until I find a better solution, I think I'll keep the image for that test.

pepicrft commented on Example/Tests/Configuration/UnsplashKitSharedExample.swift:
```diff
@@ -0,0 +1,35 @@
+                }
+                let stubbedImageData = UIImagePNGRepresentation(stubbedImage)!
+                let resultImageData = UIImagePNGRepresentation(resultImage!)!
+                expect(resultImageData.length).to(equal(stubbedImageData.length))
```
hahah what if two images have the same length? :trollface: fair enough

sergigracia replied:
I know, but I couldn't find a proper way to compare 2 images... :S 
`UIImagePNGRepresentation(fuu) == UIImagePNGRepresentation(faa)` didn't work

pepicrft replied:
Yep, I was kidding.

pepicrft commented on Example/Tests/UnsplashKitSpec.swift:
```diff
@@ -0,0 +1,23 @@
+
+@testable import UnsplashKit
+
+extension SourceRequestFactory: Equatable {}
```
I don't think this should be `Equatable` since it doesn't represent an entity.

pepicrft commented on Example/Tests/UnsplashKitSpec.swift:
```diff
@@ -0,0 +1,23 @@
+        
+        let requestFactory = SourceRequestFactory()
+                
+        describe("-init") {
```
We should never test if we set the attributes in a constructor. You can remove this test.

sergigracia replied:
but then this class will appear as untested, how do you cover this class?

pepicrft replied:
I don't think so since what the test coverage measures is how many times the code execution goes through that part of code. You construct this kit for testing so you're going through the constructor every time. @sergigracia

sergigracia replied:
You're right ;)

pepicrft commented on Example/Tests/UnsplashKit+CallsSpecs.swift:
```diff
@@ -0,0 +1,97 @@
+        var requestFactory: SourceRequestFactory!
+        var image: UIImage!
+        
+        func builder(request: NSURLRequest) -> Response {
```
👏

pepicrft replied:
I wouldn't take data from this method that is out of the scope like the image. Instead, pass the image as a parameter:

``` swift
func builder(request: NSURLRequest, image: UIImage) -> Response
```

sergigracia replied:
I cannot do that, `stub(uri:, builder:)` expects a `(NSURLRequest) -> Response` as builder.

pepicrft replied:
Some currying :trollface: 

``` swift
func builder(image: UIImage)(request: NSURLRequest) -> Response
```

Then you can do

``` swift
stub(uri: "xxx", builder: builder(image: myImage))
```

sergigracia replied:
<img width="598" alt="screen shot 2016-08-15 at 20 59 02" src="https://cloud.githubusercontent.com/assets/1061658/17675700/21d297d0-632b-11e6-83a8-cf667b36ac45.png">
XD #goodtoknow

pepicrft commented on Example/Tests/UnsplashKit+CallsSpecs.swift:
```diff
@@ -0,0 +1,97 @@
+        }
+        
+        beforeEach() {
+            subject = UnsplashKit()
```
I think `UnsplashClient` is a better name.

sergigracia replied:
agree ;)

pepicrft commented on Example/Tests/UnsplashKit+CallsSpecs.swift:
```diff
@@ -0,0 +1,97 @@
+        }
+        
+        describe("-randomPhoto") {
+            itBehavesLike("kit-call", sharedExampleContext: { () -> (NSDictionary) in
```
👏

sergigracia committed:
Add about section to the Readme

sergigracia committed:
Remove UnsplashKit Spec

sergigracia committed:
Rename UnsplasKit to UnsplashClient for public class

sergigracia committed:
Merge branch 'master' into feature/NonReactivePublicInterface

sergigracia committed:
Fixed a linter warning

Update Readme

Update Example project Podfile to use correct specs repo

Update UnplashImage class to support watchos

Update Podfile.lock and podspec

Fix import main os frameworks for different oss

sergigracia committed:
Updated pod spec support (iOS and tvOS)

sergigracia committed:
Merge branch 'master' into feature/NonReactivePublicInterface

pepibumur commented:
Good job @sergigracia !

sergigracia commented:
Thanks @pepibumur 👍
